python/ctsm/subset_data.py

```diff
@@ -0,0 +1,735 @@
+            const=True,
+            required=False,
+            default=False,
+        )
+        subparser.add_argument(
+            "--datm-syr",
+            help="Start year for creating DATM forcing at single point/region. [default: %(default)s]",
+            action="store",
+            dest="datm_syr",
+            required=False,
+            type=int,
+            default=1901,
+        )
+        subparser.add_argument(
+            "--datm-eyr",
+            help="End year for creating DATM forcing at single point/region. [default: %(default)s]",
+            action="store",
+            dest="datm_eyr",
+            required=False,
+            type=int,
+            default=2014,
+        )
+        subparser.add_argument(
+            "--crop",
+            help="Flag for creating datasets using the extensive list of prognostic crop types. [default: %(default)s]",
+            action="store",
+            dest="crop_flag",
+            type=str2bool,
+            nargs="?",
+            const=True,
+            required=False,
+            default=True,
+        )
+        subparser.add_argument(
+            "--dompft",
+            help="Dominant PFT type . [default: %(default)s] ",
+            action="store",
+            dest="dom_pft",
+            type=int,
+            default=7,
+        )
+
+        if subparser == pt_parser:
+            parser_name = "single_point"
+        else:
+            parser_name = "regional"
+
+        subparser.add_argument(
+            "--outdir",
+            help="Output directory. \n [default: %(default)s]",
+            action="store",
+            dest="out_dir",
+            type=str,
+            default=os.path.join(os.getcwd(), "subset_data_" + parser_name),
+        )
+
+    # -- print help for both subparsers
+    parser.epilog = textwrap.dedent(
+        f"""\
+         {pt_parser.format_help()}
+         {rg_parser.format_help()}
+         """
+    )
+    return parser
+
+
+def plat_type(x):
+    """
+    Function to define lat type for the parser
+    and
+    raise error if latitude is not between -90 and 90.
+
+    Args:
+        x(str): latitude
+
+    Raises:
+        Error when x (latitude) is not between -90 and 90.
+
+    Returns:
+        x (float): latitude in float
+
+    """
+    x = float(x)
+    if (x < -90) or (x > 90):
+        raise argparse.ArgumentTypeError(
+            "ERROR: Latitude should be between -90 and 90."
+        )
+    return x
+
+
+def plon_type(x):
+    """
+    Function to define lon type for the parser and
+    convert negative longitudes and
+    raise error if lon is not between -180 and 360.
+
+    Args:
+        x (str): longitude
+
+    Raises:
+        Error: when latitude is <-180 and >360.
+
+    Returns:
+        x(float): converted longitude between 0 and 360
+    """
+    x = float(x)
+    if (-180 < x) and (x < 0):
+        print("lon is :", x)
+        x = x % 360
+        print("after modulo lon is :", x)
```

Comment:
I'd rather keep these two functions for the following reasons:
-  These are functions that are written specifically for the command line arguments and will return `argparse.ArgumentTypeError` if it is not set correctly. 
- Here we also have the `plat_type` which is exactly similar to `plon_type` raising error if the bounds are not correct.  I noticed the similar does not exist in utils.py. 
-  `lon_range_0_to_360` is adapted from this function (`plot_type`) but it is not very pythonic and easy to follow. Here I used modulo %  instead of the different conditionals and this was advised to me by @jedwards4b to use modulo instead of the conditionals.

For the purpose of your comments, I removed the print and addressed your comments.

Reply:
Finally, I suggest we add args_utils.py which include the command line argument types that are going to be used for other codes. For example, plat_type, plon_type are going to be used for command line args of other codes.

Reply:
Summary from today's discussion on handling plon_type: We'll have plon_type be a wrapper that calls lon_range_0_to_360. We'll probably change the implementation of lon_range_0_to_360 to use modulo, since some others find that more clear.

We'll put plon_type and plat_type in a new arg_utils, but that could either happen in this PR or a later set of changes.

The implementation of plon_type could then look something like this:

```python
plon_out = float(plon)
if plon_out < -180 or plon_out > 360:
    raise argparse.ArgumentTypeError(...)
return lon_range_0_to_360(plon_out)
```

(Note another change I made here is to introduce a new variable for the output, since I find it confusing when the input argument is modified in-place, especially if this leads to a change in data type as it does here. I don't feel strongly about this, though I'd like to move towards the use of type hints in our python code, and I think the current implementation of changing the type of a given variable would be problematic with that.)

Reply:
This is addressed now. 
Following our discussions, I introduced `args_utils` which includes types and functions for command line args handling. I also introduced `config_utils` which has includes all the utilities and functions for reading config files.

Reply:
This is handled, arg_utils now uses lon_range_0_to_360. I like this as a solution as longitude is a common input that needs to be checked.